DephyInc/Actuator-Package
# CON-147 Convert Exo to EB5x

# Summary

The command "./manage_builder.sh Exo-rigid4.1-left" could cause some confusions since "Exo" in general refers to all Exo projects (EB5X, EB60, EB603, EB604 etc.). However, the build Exo command here only refers to EB5x projects. The changes made in this branch is to build the EB5x projects with the term "EB5x" instead of "Exo", for example, we will use "./manage_builder.sh EB5x-rigid4.1-left" command to build EB5x projects instead of the "./manage_builder.sh Exo-rigid4.1-left" command. 

[Related Flexsea-Embedded-STM PR: [CON-147_convertExoCmake]](https://github.com/DephyInc/FlexSEA-Embedded-STM/pull/77)
[Related flexsea-core PR: [CON-147_convertExoCmake]](https://github.com/DephyInc/flexsea-core/pull/42)
[Related fx_plan_stack PR: [CON-147_convertExoCmake]](https://github.com/DephyInc/fx_plan_stack/pull/85)


# Changes

* Changes variables names, structure names, macros, function names, and file names in Exo struct files (.h, .c) to use 'EB5x' references instead of 'Exo', including changing the name of "exo_struct.h/.c" file to "eb5x_struct.h/.c"

# Test Plan

 1. Make sure all exo struct files have been deleted
 2. Make sure everything still runs as it did before

# Expected Results

* Everything still builds. No functional changes.

#Requirements
- [ ] verify libs are the same as those in CON-147 branch of planStack
- [ ] Go through code and update all reference of exostruct to eb5x struct
- [ ] Run all Python scripts and make sure they still work on Linux
- [ ] Run all Python scripts and make sure they still work on windows
- [ ] Run all C-scripts and make sure they still work on linux
- [ ] Run all C-scripts and make sure they still work on windows
